pkg/controllers/workapplier/status.go

```diff
@@ -412,33 +421,41 @@ func setWorkAppliedCondition(
 		appliedCond = &metav1.Condition{
 			Type:               fleetv1beta1.WorkConditionTypeApplied,
 			Status:             metav1.ConditionFalse,
-			Reason:             notAllManifestsAppliedReason,
+			Reason:             WorkNotAllManifestsAppliedReason,
 			Message:            fmt.Sprintf(notAllManifestsAppliedMessage, appliedManifestCount, manifestCount),
 			ObservedGeneration: work.Generation,
 		}
 	}
-	meta.SetStatusCondition(&work.Status.Conditions, *appliedCond)
+
+	if appliedCond != nil {
+		meta.SetStatusCondition(&work.Status.Conditions, *appliedCond)
+	} else {
+		// For consistency reasons, Fleet will remove the Applied condition if the
+		// it is not set in the current run (i.e., ReportDiff mode is on).
+		meta.RemoveStatusCondition(&work.Status.Conditions, fleetv1beta1.WorkConditionTypeApplied)
+	}
 }
 
 // setWorkAvailableCondition sets the Available condition on a Work object.
 //
 // A Work object is considered to be available if all of its applied manifests are available.
 func setWorkAvailableCondition(
 	work *fleetv1beta1.Work,
-	manifestCount, availableManifestCount, untrackableAppliedObjectsCount int,
+	manifestCount, appliedManifestCount, availableManifestCount, untrackableAppliedObjectsCount int,
 ) {
 	var availableCond *metav1.Condition
 	switch {
 	case work.Spec.ApplyStrategy != nil && work.Spec.ApplyStrategy.Type == fleetv1beta1.ApplyStrategyTypeReportDiff:
 		// ReportDiff mode is on; no apply op has been performed, and consequently
 		// Fleet will not update the Available condition.
-		return
+	case appliedManifestCount != manifestCount:
+		// Not all manifests have been applied; skip updating the Available condition.
```

Comment:
same here, no test change needed?

Reply:
In the prev. commit there are a number of integration tests that come with the always-set `Available` condition, and it's removed in this PR -> but it's not explicit/covered enough, will add more test cases.